blockstore/src/in_memory_blockstore.rs

```diff
@@ -22,13 +21,7 @@ impl<const MAX_MULTIHASH_SIZE: usize> InMemoryBlockstore<MAX_MULTIHASH_SIZE> {
     }
 
     fn insert_cid(&self, cid: CidGeneric<MAX_MULTIHASH_SIZE>, data: &[u8]) -> Result<()> {
-        let cid_entry = self.map.entry(cid);
-        if matches!(cid_entry, Entry::Occupied(_)) {
-            return Err(BlockstoreError::CidExists);
-        }
-
-        cid_entry.insert(data.to_vec());
-
+        self.map.insert(cid, data.to_vec());
```

Comment:
CID on the bitswap protocol can represent data that are up to 2MiB. So, as an optimization, I think we shouldn't reinsert if something already exists.

Reply:
done